src/components/sidePanel/sidePanel.tsx

```diff
@@ -0,0 +1,90 @@
+import { ReactNode, useId } from 'react';
+import classNames from 'classnames/bind';
+import { CloseIcon } from '@components/icons';
+import { BaseIconButton } from '@components/baseIconButton';
+import styles from './sidePanel.module.scss';
+
+const cx = classNames.bind(styles);
+
+export interface SidePanelProps {
+  className?: string;
+  title?: ReactNode;
+  headerComponent?: ReactNode;
+  descriptionComponent?: ReactNode;
+  contentComponent?: ReactNode;
+  footerComponent?: ReactNode;
+  top?: number;
+  side?: 'left' | 'right';
+  isOpen?: boolean;
+  onClose?: () => void;
+  closeButtonAriaLabel?: string;
+}
+
+export const SidePanel = ({
+  className,
+  title,
+  descriptionComponent,
+  headerComponent,
+  contentComponent,
+  footerComponent,
+  top = 0,
+  side = 'right',
+  isOpen = true,
+  onClose,
+  closeButtonAriaLabel = 'Close panel',
+}: SidePanelProps) => {
+  const titleId = useId();
+
+  const handleClose = () => {
+    onClose?.();
+  };
+
```

Comment:
```suggestion

useEffect(() => {
  const handleEscape = (e: KeyboardEvent) => {
    if (e.key === 'Escape' && isOpen && onClose) {
      onClose();
    }
  };

  document.addEventListener('keydown', handleEscape);
  return () => document.removeEventListener('keydown', handleEscape);
}, [isOpen, onClose]);
```

To improve accessibility, we can add ESC handling to close